test/library/packages/csv/CSV.chpl

```diff
@@ -90,11 +93,44 @@ module CSV {
       }
     }
 
+    /* Read a CSV file with fields of the types given
+       in the tuple argument to the function
+     */
     iter read(type t) throws where isTupleType(t) {
       for r in read((...t)) {
         yield r;
       }
     }
+
+    /* Read a CSV file with arbitrarily many rows and columns. Returns the
+       data as strings in a 2D array. */
+    proc read(type t) throws where t == string {
+      if ch.writing {
+        throw new owned ReaderWriterMismatchError();
+      }
+      var r: t;
+      var skipHeader = hasHeader;
+
+      var lines = ch.lines();
+      var firstLine = lines[1];
+      var vals = firstLine.strip().split(sep);
+      var A: [1..lines.numElements, 1..vals.numElements] string;
+
+      if !skipHeader {
+        A[1, ..] = vals;
```

Comment:
When this is skipped, does the returned array have an initial row of 0s?

Reply:
Could be worth adding a test for this case too.

Reply:
It did have a row of empty strings.  I fixed it to be one row smaller and added a test.